app/models/measure_collection.rb

```diff
@@ -1,90 +1,56 @@
 class MeasureCollection
   include Enumerable
 
-  attr_accessor :measures, :presenter_klass
+  attr_accessor :measures
 
+  delegate :present?, to: :measures
   delegate :size, :length, :collect, :map, :each, :all?, :include?, :to_ary, to: :to_a
 
-  def initialize(measures, presenter_klass = MeasurePresenter)
+  def initialize(measures)
     @measures = measures.clone
```

Comment:
We never mutate the measures, here, so we should be able to kill the heavy lifting of cloning the individual measures at some stage